plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/StandardDeployJob.java

```diff
@@ -0,0 +1,46 @@
+package com.google.cloud.tools.eclipse.appengine.deploy.standard;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+
+import com.google.cloud.tools.eclipse.appengine.deploy.AppEngineDeployJob;
+
+public class StandardDeployJob implements AppEngineDeployJob {
+  
+  private ProjectToStagingExporter projectToStagingExporter;
+  private StagingDirectoryProvider stagingDirectoryProvider;
+  private IProject project;
+
+  public StandardDeployJob(ProjectToStagingExporter exporter, 
+                           StagingDirectoryProvider stagingDirectoryProvider,
+                           IProject project) {
+    projectToStagingExporter = exporter;
+    this.stagingDirectoryProvider = stagingDirectoryProvider;
+    this.project = project;
+  }
+
+  @Override
+  public void run(IProgressMonitor monitor) throws CoreException {
+    String stageDir = stagingDirectoryProvider.getStagingDirectory();
```

Comment:
For now I'll leave it as it is until in the next steps we implement a better way to configure it.